### PR TITLE
add Document Information Extraction API client (continued)

### PR DESCRIPTION
**Pull request by tillgeissler**
This PR adds the DoxApiClient class to access the SAP Document Information Extraction REST API.

(It is a continuation of [this](https://github.com/SAP/business-document-processing/pull/29) pull request.)

### CODE REVIEW

File: sap_business_document_processing/document_classification_client/dc_api_client.py [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+                            log_msg_after=f'Successfully got the information about the documents in the '
+                            f'dataset {dataset_id}')
+        return response.json()
+
+    def get_classification_documents_info(self, model_name, model_version):
+        """
+        Gets the information about recently classified documents
+        :param model_name: The name of the model against which the documents were classified
+        :param model_version: The version of the model against which the documents were classified
+        :return: An array of document information, each document information includes its reference ID and the
```
Comment (lextiz) [inline sap_business_document_processing/document_classification_client/dc_api_client.py:222]: Please see the https://github.com/SAP/business-document-processing/pull/29#discussion_r613941446

File: sap_business_document_processing/document_information_extraction_client/dox_api_client.py [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+    DATA_ACTIVATION_ID_ENDPOINT, DATA_ENDPOINT, DATA_ASYNC_ENDPOINT, DATA_ID_ENDPOINT, DOCUMENT_ENDPOINT, \
+    DOCUMENT_CONFIRM_ENDPOINT, DOCUMENT_ID_ENDPOINT, DOCUMENT_ID_REQUEST_ENDPOINT, DOCUMENT_PAGE_ENDPOINT, \
+    DOCUMENT_PAGE_DIMENSIONS_ENDPOINT, DOCUMENT_PAGES_DIMENSIONS_ENDPOINT, DOCUMENT_PAGE_TEXT_ENDPOINT, \
+    DOCUMENT_PAGES_TEXT_ENDPOINT
+from .helpers import create_document_options, create_capability_mapping_options
+
+
+class DoxApiClient(CommonClient):
+    """
+    This class is the main entry point to consume the SAP Document Information Extraction REST API from a Python
```
Comment (lextiz) [inline sap_business_document_processing/document_information_extraction_client/dox_api_client.py:29]: Where did we put the link to the documentation? (see: https://github.com/SAP/business-document-processing/pull/29#discussion_r613980446)

**Review by lextiz — COMMENTED** [review]
The fixes look very good in general, lets complete the following and we can merge, release the package (PyPI and release notes) and close the topic:

- Either the shell script that uploads the package to PyPI needs to be adapted or we need an alternative to it
- Let's please add type annotations to the DC client as well
- Notebooks review in human-readable format (preferably together with PO) is pending
- Please regenerate the API docs and the Jupiter notebooks outputs
- We will need to review the release notes, when ready

File: requirements.txt [hunk 3]
```diff
@@ -3,3 +3,5 @@ numpy==1.17.2
 matplotlib==3.1.1
 scikit-learn==0.21.3
 pandas==0.25.1
+fpdf==1.7.2
```
Comment (lextiz) [inline requirements.txt]: Why both are used?

**Review by lextiz — COMMENTED** [review]
@tillgeissler Looks good, could you please comment on the unresolved remarks?

File: requirements.txt (hunk 3 above)
Comment (tillgeissler) [reply requirements.txt]: the PDF dependencies have been fixed and only fpdf2 used

File: sap_business_document_processing/document_classification_client/dc_api_client.py (hunk 1 above)
Comment (tillgeissler) [reply sap_business_document_processing/document_classification_client/dc_api_client.py]: Has been added

File: sap_business_document_processing/document_information_extraction_client/dox_api_client.py (hunk 2 above)
Comment (tillgeissler) [reply sap_business_document_processing/document_information_extraction_client/dox_api_client.py]: Has been added with a later commit

File: examples/start_testing_merged_client_library.ipynb [hunk 4]
```diff
@@ -0,0 +1,316 @@
+{
```
[the remark this answers is not in the document]
Comment (tillgeissler) [reply examples/start_testing_merged_client_library.ipynb]: Has been removed, as we don't need use it

**Review by lextiz — APPROVED** [review]

**Pull request merged**